# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/chromium/hotfix-rustc-t2.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- chromium-136.0.7103.33/build/config/rust.gni.vanilla	2025-04-19 11:06:31.141505483 +0200
+++ chromium-136.0.7103.33/build/config/rust.gni	2025-04-19 11:08:43.032008193 +0200
@@ -175,13 +175,13 @@
 rust_abi_target = ""
 if (is_linux || is_chromeos) {
   if (current_cpu == "arm64") {
-    rust_abi_target = "aarch64-unknown-linux-gnu"
+    rust_abi_target = "aarch64-t2-linux-gnu"
     cargo_target_abi = ""
   } else if (current_cpu == "x86") {
-    rust_abi_target = "i686-unknown-linux-gnu"
+    rust_abi_target = "i686-t2-linux-gnu"
     cargo_target_abi = ""
   } else if (current_cpu == "x64") {
-    rust_abi_target = "x86_64-unknown-linux-gnu"
+    rust_abi_target = "x86_64-t2-linux-gnu"
     cargo_target_abi = ""
   } else if (current_cpu == "arm") {
     if (arm_float_abi == "hard") {
@@ -201,21 +201,21 @@
       # targets in fact target Thumb, see:
       # https://github.com/rust-lang/rust/issues/44722
       if (arm_use_neon) {
-        rust_abi_target = "thumbv7neon-unknown-linux-gnueabi" + float_suffix
+        rust_abi_target = "thumbv7neon-t2-linux-gnueabi" + float_suffix
       } else {
-        rust_abi_target = "armv7-unknown-linux-gnueabi" + float_suffix
+        rust_abi_target = "armv7-t2-linux-gnueabi" + float_suffix
       }
       cargo_target_abi = "eabi" + float_suffix
     } else {
-      rust_abi_target = "arm-unknown-linux-gnueabi" + float_suffix
+      rust_abi_target = "arm-t2-linux-gnueabi" + float_suffix
       cargo_target_abi = "eabi" + float_suffix
     }
   } else if (current_cpu == "riscv64") {
-    rust_abi_target = "riscv64gc-unknown-linux-gnu"
+    rust_abi_target = "riscv64gc-t2-linux-gnu"
     cargo_target_abi = ""
   } else {
     # Best guess for other future platforms.
-    rust_abi_target = current_cpu + "-unknown-linux-gnu"
+    rust_abi_target = current_cpu + "-t2-linux-gnu"
     cargo_target_abi = ""
   }
 } else if (is_android) {
